From c1b834f80311706db2b5070cbccdcba3aacc90e5 Mon Sep 17 00:00:00 2001
From: Kazuhito Hagio <k-hagio@ab.jp.nec.com>
Date: Tue, 23 Jul 2019 11:50:52 -0400
Subject: [PATCH 1/2] Do not proceed when get_num_dumpable_cyclic() fails

Currently, when get_num_dumpable_cyclic() fails and returns FALSE in
create_dump_bitmap(), info->num_dumpable is set to 0 and makedumpfile
proceeds to write a broken dumpfile slowly with incorrect progress
indicator due to the value.

It should not proceed when get_num_dumpable_cyclic() fails.

Signed-off-by: Kazuhito Hagio <k-hagio@ab.jp.nec.com>
---
 makedumpfile.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/makedumpfile.c b/makedumpfile.c
index d76a4358a2b2..88df788fd6e7 100644
--- a/makedumpfile-1.6.6/makedumpfile.c
+++ b/makedumpfile-1.6.6/makedumpfile.c
@@ -7344,7 +7344,8 @@ create_dump_bitmap(void)
 		if (!prepare_bitmap2_buffer())
 			goto out;
 
-		info->num_dumpable = get_num_dumpable_cyclic();
+		if (!(info->num_dumpable = get_num_dumpable_cyclic()))
+			goto out;
 
 		if (!info->flag_elf_dumpfile)
 			free_bitmap2_buffer();
@@ -7362,7 +7363,8 @@ create_dump_bitmap(void)
 		if (!create_2nd_bitmap(&cycle))
 			goto out;
 
-		info->num_dumpable = get_num_dumpable_cyclic();
+		if (!(info->num_dumpable = get_num_dumpable_cyclic()))
+			goto out;
 	}
 
 	ret = TRUE;
-- 
2.18.1

